nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -439,30 +470,30 @@ void RegulatedPurePursuitController::applyConstraints(
   }
 
   // Use the lowest of the 2 constraint heuristics, but above the minimum translational speed
-  linear_vel = std::min(cost_vel, curvature_vel);
```

Comment:
I think in this function this might work, but you should potentially try to slow down a little and make this more human understandable. You get the sign in the main function, apply it to the velocity, and then feed it back into here where you end up doing a bunch of other mechanics with fabs and whatnot. Would there not just be a single place that when you detect a change in direction you could flip the sign? Or a flag you could pass around to functions / areas of code that need it to let it know that its sign is flipped?

Reply:
It would also be good to verify yourself that you indeed do set the signs correctly so that the heuristics and the collision checker are using the correct velocity signed direction (since that will be important for the collision checker to be checking in the right direction of travel)